composite/dns-record-check/src/adapter.ts

```diff
@@ -0,0 +1,34 @@
+import { Execute, DNSResponseAnswer, AdapterRequest, DNSResponse } from '@chainlink/types'
+import { Validator } from '@chainlink/external-adapter'
+import DNS from '@chainlink/dns-query'
+import { Requester } from '@chainlink/external-adapter'
+
+const inputParams = {
+  record: true,
+  ...DNS.inputParams,
+}
+
+export const execute: Execute = async (input) => {
```

Comment:
Same story here, we don't expose the raw function, but a factory to build the function.

```suggestion
const execute: ExecuteWithConfig = async (input, config) => {
```

```javascript
export const makeExecute: ExecuteFactory = (config?: Config) => (input) => execute(input, config || makeConfig())
```

The question is what is the required `Config` type? Well in this case it is the config of the underlying `dns-query` adapter, and nothing else as it looks like this adapter does not require any specific configuration. It just needs to configure the underlying adapter.